src/services/dexScaner/UniswapPoolScanner.ts

```diff
@@ -0,0 +1,96 @@
+import {inject, injectable} from 'inversify';
+
+import Settings from '../../types/Settings.js';
+import {PoolCreatedEvent, UniswapV3Factory} from '../../blockchains/evm/contracts/UniswapV3Factory.js';
+import {UniswapPoolService} from '../uniswap/UniswapPoolService.js';
+import {BlockchainScanner} from '../BlockchainScanner.js';
+import {UniswapV3Helper} from '../../blockchains/evm/contracts/UniswapV3Helper.js';
+import {DexPoolScannerService} from './DexPoolScannerService.js';
+
+export type UniswapPoolCreatedEvent = {
+  token0: string;
+  token1: string;
+  fee: bigint;
+  pool: string;
+  anchor: number;
+};
+
+@injectable()
+export class UniswapPoolScanner extends BlockchainScanner {
+  readonly settings: Settings;
+  readonly markerId = 'UniswapPoolScanner';
+
+  @inject(UniswapPoolService) uniswapPoolService!: UniswapPoolService;
+  @inject(DexPoolScannerService) dexPoolScannerService!: DexPoolScannerService;
+  @inject(UniswapV3Factory) uniswapV3Factory!: UniswapV3Factory;
+  @inject(UniswapV3Helper) uniswapV3Helper!: UniswapV3Helper;
+
+  constructor(@inject('Settings') settings: Settings) {
+    super();
+
+    this.settings = settings;
+  }
+
+  async apply(fromBlock: number, toBlock: number): Promise<boolean> {
+    if (!this.chainId) {
+      this.logger.info('[UniswapPoolScanner] chainId not set');
+      return false;
+    }
+
+    this.uniswapV3Factory.setProvider(this.chainId);
```

Comment:
use repository, not seter.

Reply:
Let me know if the pattern is better. I created the EthereumUniswapV3 class
For new DexProtocol, we would need to create a new mapping, not sure if it is intended tbh